### DISCUSSION TITLE
5.15内核upnp无效怎样破？

### DISCUSSION THREAD

**Discussion by he110777** [Boos4721/openwrt#130] [Bugs/问题反馈]
5.10的upnp群晖等能正常映射，5.15内核就无效。5月6日更新后AX6已经1天没有重启了，编译是选的是Shortcut-FE。

**Comment by Boos4721** [comment]
SEEMS WORKS

**Reply by he110777** [reply]
在5.15里upnp是没有活动的重定向呢。

**Reply by he110777** [reply]
May  8 23:30:38 2022 daemon.info miniupnpd[23409]: system uptime is 4372 seconds
Sun May  8 23:30:38 2022 daemon.info miniupnpd[23409]: mnl_socket bound, port_id=23409
Sun May  8 23:30:38 2022 daemon.err miniupnpd[23409]: send_batch: mnl_cb_run returned -1
Sun May  8 23:30:38 2022 daemon.err miniupnpd[23409]: table_op(0, 1, miniupnpd) send_batch failed -4
Sun May  8 23:30:38 2022 daemon.err miniupnpd[23409]: Failed to init redirection engine. EXITING
Sun May  8 23:30:40 2022 daemon.info miniupnpd[23713]: system uptime is 4375 seconds
Sun May  8 23:30:40 2022 daemon.info miniupnpd[23713]: mnl_socket bound, port_id=23713
Sun May  8 23:30:40 2022 daemon.err miniupnpd[23713]: send_batch: mnl_cb_run returned -1
Sun May  8 23:30:40 2022 daemon.err miniupnpd[23713]: table_op(0, 1, miniupnpd) send_batch failed -4
Sun May  8 23:30:40 2022 daemon.err miniupnpd[23713]: Failed to init redirection engine. EXITING

**Reply by he110777** [reply]
https://forum.openwrt.org/t/solved-miniupnp-error-on-openwrt-snapshot/124930

**Reply by Boos4721** [reply]
> [forum.openwrt.org/t/solved-miniupnp-error-on-openwrt-snapshot/124930](https://forum.openwrt.org/t/solved-miniupnp-error-on-openwrt-snapshot/124930)

**THANKS FOR UR ADVISE, FIX NOW!**

**Reply by he110777** [reply]
我不能确定这个方法是否有效，但是5.10编译的luci-app-upnp带的依赖是miniupnpd，而5.15编译的是miniupnpd-nftables，连接所说的miniupnpd-iptables我也不清楚这3个有什么不同。

**Reply by Boos4721** [reply]
> 我不能确定这个方法是否有效，但是5.10编译的luci-app-upnp带的依赖是miniupnpd，而5.15编译的是miniupnpd-nftables，连接所说的miniupnpd-iptables我也不清楚这3个有什么不同。

WORKS

**Reply by ghost** [reply]
经测试UPNP已生效，感谢两位！

**Reply by Gaojianli** [reply]
经过测试，opkg安装miniupnpd-iptables可以恢复upnp
